app/controllers/api/assignments_controller.rb

```diff
@@ -18,7 +18,8 @@ class AssignmentsController < MainApiController
     # Returns a list of assignments and their attributes
     # Optional: filter, fields
     def index
-      assignments = get_collection(current_course.assessments) || return
+      assignments = get_collection(current_role&.visible_assessments
```

Comment:
This query is too permissive, though possibly related to my comment below on the policy. `current_role` should never be `nil` if the authorization was successful. Also, the `where` clause is not necessary because the `visible_assessments` method should take into account the associated `course` of the current role.